### PR TITLE
Fixing `snap-preact` Config for Services

### PR DESCRIPTION
**Pull request by korgon**
* fully wiring config services
* add configuration to `deepmerge` to prevent loss of special object prototypes when passed via config (eg: services like UrlTranslator)

### CODE REVIEW

File: packages/snap-preact/src/create/createAutocompleteController.ts [hunk 1]
```diff
@@ -14,7 +14,10 @@ import type { SnapControllerServices, SnapAutocompleteControllerConfig } from '.
 configureMobx({ useProxies: 'never' });
 
 export default (config: SnapAutocompleteControllerConfig, services?: SnapControllerServices): AutocompleteController => {
-	const urlManager = services?.urlManager || new UrlManager(new UrlTranslator(config.url), reactLinker).detach();
+	console.log('AC services?', services);
```
Comment (dkonieczek) [inline packages/snap-preact/src/create/createAutocompleteController.ts:17]: 👀

File: packages/snap-preact/src/create/createAutocompleteController.ts [hunk 2]
```diff
@@ -14,7 +14,10 @@ import type { SnapControllerServices, SnapAutocompleteControllerConfig } from '.
...
+	const urlManager = (services?.urlManager || new UrlManager(new UrlTranslator(config.url), reactLinker)).detach();
+
+	console.log('URLAMANATERER! - ', urlManager, urlManager.subscribe);
```
Comment (dkonieczek) [inline packages/snap-preact/src/create/createAutocompleteController.ts:20]: 👀

**Review by dkonieczek — CHANGES_REQUESTED** [review]

File: packages/snap-preact/src/create/createAutocompleteController.ts (hunk 1 above)
Comment (korgon) [reply packages/snap-preact/src/create/createAutocompleteController.ts:17]: Derp. Trying to go too fast...

File: packages/snap-preact/src/create/createAutocompleteController.ts (hunk 2 above)
Comment (korgon) [reply packages/snap-preact/src/create/createAutocompleteController.ts:20]: Derp. Trying to go too fast...

**Review by dkonieczek — APPROVED** [review]
